pkg/virt-api/webhooks/validating-webhook/admitters/vms-admitter.go

```diff
@@ -506,24 +516,9 @@ func (admitter *VMsAdmitter) validateVolumeRequests(vm *v1.VirtualMachine) ([]me
 			}
 
 			// Validate the disk is configured properly
-			if volumeRequest.AddVolumeOptions.Disk == nil {
-				return []metav1.StatusCause{{
-					Type:    metav1.CauseTypeFieldValueInvalid,
-					Message: fmt.Sprintf("AddVolume request for [%s] requires the disk field to be set.", name),
-					Field:   k8sfield.NewPath("Status", "volumeRequests").String(),
-				}}, nil
-			} else if volumeRequest.AddVolumeOptions.Disk.DiskDevice.Disk == nil {
-				return []metav1.StatusCause{{
-					Type:    metav1.CauseTypeFieldValueInvalid,
-					Message: fmt.Sprintf("AddVolume request for [%s] requires diskDevice of type 'disk' to be used.", name),
-					Field:   k8sfield.NewPath("Status", "volumeRequests").String(),
-				}}, nil
-			} else if volumeRequest.AddVolumeOptions.Disk.DiskDevice.Disk.Bus != "scsi" {
-				return []metav1.StatusCause{{
-					Type:    metav1.CauseTypeFieldValueInvalid,
-					Message: fmt.Sprintf("AddVolume request for [%s] requires disk bus to be 'scsi'. [%s] is not permitted", name, volumeRequest.AddVolumeOptions.Disk.DiskDevice.Disk.Bus),
-					Field:   k8sfield.NewPath("Status", "volumeRequests").String(),
-				}}, nil
+			invalidDiskStatusCause, ok := validateDiskConfiguration(volumeRequest.AddVolumeOptions.Disk, name)
```

Comment:
You shouldn't need the `bool` here, just check the length of `invalidDiskStatusCause`